internal/hwmon/hwmon_test.go

```diff
@@ -76,3 +79,155 @@ func TestFindPlatform(t *testing.T) {
 	// THEN
 	assert.Equal(t, "", platform)
 }
+
+func TestUpdateFanConfigFromHwMonControllers(t *testing.T) {
+	var tests = []struct {
+		tn            string
+		hwMonConfigs  []configuration.HwMonFanConfig
+		hwMonPlatform string
+		configConfig  configuration.HwMonFanConfig
+		wantConfig    *configuration.HwMonFanConfig
+		wantErr       string
+	}{{
+		tn: "index config",
+		hwMonConfigs: []configuration.HwMonFanConfig{
+			{
+				Index:      1,
+				Channel:    2,
```

Comment:
I like these `Channel` and `PwmChannel` parameters, however I feel like the naming can be confusing, because the `Channel` should really be called `RpmChannel`. This of course has the downside of beeing unintuitive to use because it would seem to imply that both RpmChannel and PwmChannel are required to have both RPM readings as well as PWM control.

Do you have thoughts on that?

Reply:
From Linux's perspective, there are no fans. Only sensors (inputs) and actuators (outputs) connected to channels (think chip GPIO pin, or SPI address, etc). So fan2go is creating its abstraction for a fan. From that perspective, a fan should indeed have sensors, and the most common ones are going to be "RPM" and "PWM". A sensor is going to have a channel, and sensors _of different types on a given chip_ can share the same channel number (e.g. you can have RPM and PWM on channel "1", but never 2 RPMs on channel "1", for a given hwmon chip).

Anyways, all this to say that I would not mind using explicit names and renaming `Channel` to `RPMChannel` is probably a good change.

I think there is a bit of confusion with matching the config against the runtime fans. With the patch as it stands, you can use "Index" or "Channel" to match, but that is indeed the RPM channel. There is no way to match using the PWM channel. And this mostly stems from fan2go kind of using the RPM channel/sensor as the "primary channel" (or identity) of its notion of a fan.

But of course, not all fans have a RPM sensor (most do though, but cheapo 2 pin fans do not) and not all fans are under the control of a PWM, and a given PWM may control more than one fan, and a given RPM sensor may in fact be associated with more than one fan (avg/max sensor). Fan2go can largely ignore all that complexity IMO and only have enough complexity to be useful to its users :p